apps/backend/src/modules/onboarding/onboarding.repository.ts

```diff
@@ -0,0 +1,157 @@
+import { RelationName } from "~/libs/enums/enums.js";
+import { DatabaseTableName } from "~/libs/modules/database/database.js";
+import { type Repository } from "~/libs/types/types.js";
+
+import { type OnboardingAnswerDto } from "./libs/types/types.js";
+import { OnboardingAnswerEntity } from "./onboarding-answer.entity.js";
+import { type OnboardingAnswerModel } from "./onboarding-answer.model.js";
+import { type OnboardingQuestionModel } from "./onboarding-question.model.js";
+
+class OnboardingRepository implements Repository {
+	private onboardingAnswerModel: typeof OnboardingAnswerModel;
+
+	private onboardingQuestionModel: typeof OnboardingQuestionModel;
+
+	public constructor(
+		onboardingAnswerModel: typeof OnboardingAnswerModel,
+		onboardingQuestionModel: typeof OnboardingQuestionModel,
+	) {
+		this.onboardingAnswerModel = onboardingAnswerModel;
+		this.onboardingQuestionModel = onboardingQuestionModel;
+	}
+
+	public async countQuestions(): Promise<number> {
+		return await this.onboardingQuestionModel.query().resultSize();
```

Comment:
Can we use `.count` here?

Reply:
Not that it is not suitable, as far as I know `count` doesn't fetch everything from the DB, that's why I prefer it